site/src/Consumer/EmailNotificationConsumer.php

```diff
@@ -12,26 +14,105 @@
  */
 class EmailNotificationConsumer extends AbstractNotificationConsumer
 {
+    /**
+     * @var UserManagerInterface
+     */
+    protected $userManager;
+
+    /**
+     * @var \Swift_Mailer
+     */
+    protected $mailer;
+
+    /**
+     * @var Environment
+     */
+    protected $twigEnvironment;
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
     /**
      * EmailNotificationConsumer constructor
      *
-     * @param LoggerInterface $logger
+     * @param LoggerInterface      $logger
+     * @param UserManagerInterface $userManager
+     * @param Environment          $twigEnvironment
+     * @param \Swift_Mailer        $mailer
+     * @param ContainerInterface   $container
      */
-    public function __construct(LoggerInterface $logger)
-    {
+    public function __construct(
+        LoggerInterface $logger,
+        UserManagerInterface $userManager,
+        Environment $twigEnvironment,
+        ContainerInterface $container
+    ) {
         parent::__construct($logger);
+        $this->userManager = $userManager;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->container = $container;
+        $this->mailer = $container->get(
+            'swiftmailer.mailer.notification_mailer'
+        );
+    }
+
+    protected function shouldHandle()
+    {
+        $type = $this->notification->getType();
+
+        $user = $this->userManager->findUserByUsername($this->notification->getRecipient()->getUsername());
+
+        $emailSubscription = $user->isSubscribedToEmailNotifications();
+
+        if ($emailSubscription) {
+            switch ($type) {
+                case "site_information":
+                    $this->logger->info('Processing the notification and sending an email...');
+
+                    return true;
+                default:
+                    $this->logger->error('Something went wrong. Please check the notification type');
+
+                    return false;
+            }
+        }
+
+        return false;
     }
 
     /**
      * Handles an email Notification
      *
      * @return bool
+     *
+     * @throws \Exception
      */
     protected function handle(): bool
     {
-        // TODO: Process the Notification($this->notification)
         echo "Email Notification Consumer";
 
+        $userEmail = $this->userManager->findUserByUsername($this->notification->getRecipient()->getUsername());
+
+        $message = (new \Swift_Message($this->notification->getSubject()))
+            // Set sender's email from a yaml file
+            ->setFrom($this->container->getParameter(
+                'swiftmailer.mailer.default.transport.smtp.username'
```

Comment:
No. Don't use the SMTP username as sender, use a dedicated setting to get the name and email address of the sender.

Reply:
This comes from `dev.vagrant.secrets.yml` and is fetched by container in the parameters